### PR TITLE
Improve the breakpoints display while we're indexing

### LINKED ISSUE #2703: The breakpoint hits tooltip should show we're indexing
**Description:**

At the moment, the breakpoint hits tooltip shows `...` when we're indexing the replay which can make it seem like fetching the number of hits can be slow when in fact we're just waiting for the recording to finish processing.

Tasks
- [ ] Update the tooltip to say "Indexing"
- [ ] Show the panel when we're registering breakpoints
- [ ] Replace the light blue breakpoint color with the normal blue

### PR DESCRIPTION
**Pull request by hbenl**
Fixes #2703

### CODE REVIEW

File: src/devtools/client/debugger/src/actions/sources/newSources.js [hunk 1]
```diff
@@ -86,6 +86,11 @@ function checkPendingBreakpoints(cx, sourceId) {
       return;
     }
 
+    for (const bp of pendingBreakpoints) {
+      const line = bp.location.line;
+      dispatch({ type: "SET_REQUESTED_BREAKPOINT", location: { sourceId, line } });
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/actions/sources/newSources.js:91]: why not include the column as well?

File: src/devtools/client/debugger/src/selectors/visibleColumnBreakpoints.js [hunk 2]
```diff
@@ -108,7 +109,12 @@ export function getColumnBreakpoints(positions, breakpoints, viewport, selectedS
   // - the position is in the current viewport
   // - there is atleast one other breakpoint on that line
   // - there is a breakpoint on that line
-  const breakpointMap = groupBreakpoints(breakpoints, selectedSource);
+  const allBreakpoints = [...breakpoints, ...requestedBreakpoints];
+  let positions = uniqBy(
+    allBreakpoints.map(bp => bp.location),
+    ({ sourceId, line }) => `${sourceId}:${line}`
+  );
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/selectors/visibleColumnBreakpoints.js:116]: should we update the comment above to describe what is happening here?

File: src/devtools/client/debugger/src/actions/sources/newSources.js (hunk 1 above)
Comment (hbenl) [reply src/devtools/client/debugger/src/actions/sources/newSources.js:91]: I basically copied this from the only other place where a `SET_REQUESTED_BREAKPOINT` action is dispatched (in `modify.js`)

File: src/devtools/client/debugger/src/selectors/visibleColumnBreakpoints.js (hunk 2 above)
Comment (hbenl) [reply src/devtools/client/debugger/src/selectors/visibleColumnBreakpoints.js:116]: What about:
```
// collect all breakpoint positions but make sure that we don't have 2 positions for the same line in the same source
```

Comment (jasonLaster) [reply src/devtools/client/debugger/src/selectors/visibleColumnBreakpoints.js:116]: works for me

File: src/devtools/client/debugger/src/actions/sources/newSources.js (hunk 1 above)
Comment (jasonLaster) [reply src/devtools/client/debugger/src/actions/sources/newSources.js:91]: weird - feels like an omission, but tough to say...

Comment (hbenl) [reply src/devtools/client/debugger/src/actions/sources/newSources.js:91]: It does, but I found this is actually enforced [here](https://github.com/RecordReplay/devtools/blob/b4a2ce306178b3f1eb5b4d420b9dc18bfdf6abcb/src/devtools/client/debugger/src/reducers/breakpoints.js#L77)

**Comment by hbenl** [comment]
I had a hard time creating recordings that show the effect of this PR - it depends very much on the timings between backend and frontend and the frontend performance is quite different when it's being recorded. So the following recordings may not differ much visually, but the comments show what's going on under the hood:
Without this PR: https://replay.io/view?id=243df849-ac85-4216-9663-65fcc62d9317
With this PR: https://replay.io/view?id=9628363e-be9b-4a26-8fbb-61cfd0ad29ac

File: src/devtools/client/debugger/src/actions/sources/newSources.js (hunk 1 above)
Comment (jasonLaster) [reply src/devtools/client/debugger/src/actions/sources/newSources.js:91]: weird...

**Comment by jasonLaster** [comment]
Looks good. Feel free to land when the tests are green

**Review by jasonLaster — APPROVED** [review]

**Comment by hbenl** [comment]
All test are green locally

**Pull request merged**